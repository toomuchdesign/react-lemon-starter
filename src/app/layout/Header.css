@import "./layout.variables";

:root {
  --Header-height:                  60px;
  --Header-logo-height:             30px;
  --Header-logo-width:              132px;
}

.Header {
  z-index: var(--zindex--header);

  &-logo {
    float: left;
  }
}
