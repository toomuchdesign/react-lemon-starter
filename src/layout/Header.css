@import "./layout.variables";

.Header {
  width: 100%;
  height: var(--Header-height);

  z-index: var(--zindex--header);
  text-align: left;
  background: #2d3e50;
  color: #fff;

  @media (--lg) {
    position: fixed;
    left: 0;
    top: 0;
  }

  &-inner {
    height: var(--Header-height);
    margin: 0 auto;
    padding: 0 25px;
    position: relative;
  }

  &-logo {
    height: var(--Header-logo-height);
    width: var(--Header-logo-width);
    overflow: hidden;
    float: left;
    margin: calc( calc( var(--Header-height) - var(--Header-logo-height) ) / 2 ) 0;

    >a {
      display: block;
    }

    &-img {
      width: 100%;
    }
  }

  &-nav {
    float: right;

    &-list {
      display: inline-block;
      padding: 0;
      margin: 0;
    }

    &-item {
      display: inline-block;
      zoom: 1;
      vertical-align: middle;
    }

    &-link {
      line-height: var(--Header-height);
      text-decoration: none;
      color: #fff;
      padding: .5em 1em;
    }
  }
}
