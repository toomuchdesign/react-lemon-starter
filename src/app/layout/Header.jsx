import React from 'react';
import { Link } from 'react-router';
import logo from './logo.svg';
import './Header.css';

const headerMenuItems = [
  [{ name: 'about', params: { foo: 'bar' } }, 'to about params'],
  [{ name: 'about', query: { foo: 'bar' } }, 'to about query'],
];

export default function Header() {
  return (
    <div className="Header">
      <div className="Header-container">
        <div className="Header-logo">
          <Link
            to="/"
          >
            <img
              alt="Logo"
              className="Header-logo-img"
              src={logo}
            />
          </Link>
        </div>
        <div className="Header-nav">
          <ul className="Header-menu">
            {headerMenuItems.map((item, index) => (
              <li
                className="Header-item"
                key={index}
              >
                <Link
                  to={item[0]}
                >
                  {item[1]}
                </Link>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
}

Header.displayName = 'Header';
