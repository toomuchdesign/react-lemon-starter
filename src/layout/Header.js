import React from 'react';
import { Link } from 'react-router-dom';
import logo from './logo.svg';
import './Header.css';

const headerMenuItems = [
  [{ pathname: '/about/bar' }, 'to about params'],
  [{ pathname: '/about', search: '?foo=bar' }, 'to about query'],
];

export default function Header() {
  return (
    <div className="Header">
      <div className="Header-inner">
        <div className="Header-logo">
          <Link
            to="/"
          >
            <img
              alt="Logo"
              className="Header-logo-img"
              src={logo}
            />
          </Link>
        </div>
        <div className="Header-nav">
          <ul className="Header-nav-list">
            {headerMenuItems.map((item, index) => (
              <li
                className="Header-nav-item"
                key={index}
              >
                <Link
                  to={item[0]}
                  className="Header-nav-link"
                >
                  {item[1]}
                </Link>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
}

Header.displayName = 'Header';
