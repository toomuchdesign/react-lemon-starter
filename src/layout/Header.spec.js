/* eslint comma-dangle: 0 */
import React from 'react';
import {shallow} from 'enzyme';
import Header from './Header';

describe('Header default output', () => {
  it('Should output with all attributes defined', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper).toMatchSnapshot();
  });
});
