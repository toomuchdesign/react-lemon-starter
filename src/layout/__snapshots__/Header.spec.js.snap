// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header default output Should output with all attributes defined 1`] = `
<div
  className="Header"
>
  <div
    className="Header-inner"
  >
    <div
      className="Header-logo"
    >
      <Link
        replace={false}
        to="/"
      >
        <img
          alt="Logo"
          className="Header-logo-img"
          src={Object {}}
        />
      </Link>
    </div>
    <div
      className="Header-nav"
    >
      <ul
        className="Header-nav-list"
      >
        <li
          className="Header-nav-item"
          key="0"
        >
          <Link
            className="Header-nav-link"
            replace={false}
            to={
              Object {
                "pathname": "/about/bar",
              }
            }
          >
            to about params
          </Link>
        </li>
        <li
          className="Header-nav-item"
          key="1"
        >
          <Link
            className="Header-nav-link"
            replace={false}
            to={
              Object {
                "pathname": "/about",
                "search": "?foo=bar",
              }
            }
          >
            to about query
          </Link>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
